animalkingdom/src/animalkingdom/Main.java

```diff
@@ -0,0 +1,115 @@
+package animalkingdom;
+
+import java.util.*;
+// WARNING: this imports the WHOLE package, in the future IDE's will import only what we are using. importing the whole package can lead to a larger file size than needed.
+
+
+// https://www.tutorialspoint.com/java/java_constructors.htm 
+// above is just a random resource i used to help learn things
+
+public class Main
+// every java app must have public 'Main' class. the classname (in this case Main), must relate back to the file name.
+{
+	public static void renderAnimal(ArrayList<AnimalActions> animals, CheckAnimal tester)
+	// see line 14 - 16 for public static void
+	// the strings we are passing in here is an array from AnimalActions containing animals
+	// the args we are passing is CheckAnimal, to test it and see if its an animal or not (true/false)
+	{
+		for (AnimalActions v : animals )
+		// for loop itterates over a range of values > https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html
+		// takes in the following (initilization; termination; increments) { statement/s }
+		// the use of : is acting as a syntax sugar, its another way of a for-if-else loop
+		// see > https://www.sitepoint.com/java-ternary-operator/ and https://stackoverflow.com/a/2399595
+		{
+			if (tester.test(v)){
+				System.out.println(x.toString());
+				// x.toString is explained in mammels.java line 59 - 63
+			}
+		};
+	}
+
+	public static ArrayList<AnimalActions> sortAnimalsArray(ArrayList<AnimalActions> animals, CheckAnimal tester)
+	{
+		ArrayList<AnimalActions> sortedAnimalsActionsArray = new ArrayList<AnimalActions>();
+		for (AnimalActions v : animals) {
+			if (tester.test(v)){
+				sortedAnimalsActionsArray.add(v);
+				// sortedAnimalsActionsArray.add(x) is explained on line 70;
+			}
+		}
+		return sortedAnimalsActionsArray();
+		// returns the now sorted array
+	}
+	
```

Comment:
Line 40 - Don't invoke sortedAnimalsActionsArray (which is not a function, but a variable) in your return statement - the array list has already been sorted anyway.